guides/svg/template.md

```diff
@@ -0,0 +1,228 @@
+
+### A basic model of container component :
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+
+    <g transform="translate(5,5)">
+        <svg id="icon-{{id}}" class="component-icon" width="32" height="32"></svg>
+    
+        <g transform="translate(35,0)">
+            <text class="component-name" y="1em">{{ name }}</text>
+            <text class="component-type" y="2.5em">{{ definition.type }}</text>
+        </g>
+    </g>
+
+    <g transform="translate(0,50)">
+        <svg class="component-container" overflow="visible">
+            <rect class="container-background" width="100%" height="100%" fill="whitesmoke" />
+        </svg>
+    </g>
+    
+    {% if hasError %}
+    <g class="component-error">
+        <svg width="15" height="15" x="200" y="10" viewBox="0 0 100 100">
+          <circle fill="red" r="50" cx="50" cy="50"></circle>
+        </svg>
+    </g>
+    {% endif %}
+</svg>
+```
+
+### Why I see TWIG syntax in this model ?
+
+As you can see, we have some `twig` syntax in the template svg, **but** that's **not** `twig`. To render the template
+we use the `nunjucks` templating engine, you don't really need to understand what's going on here, but if you want to
+know more about it, you can read the [nunjucks documentation](https://mozilla.github.io/nunjucks/templating.html).
+
+___
+
+## Step by step
+
+### 1. Create the template :
+
+First you have to create the template, this is a `<svg></svg>` with a class `template`, we also need to define the width
+and height for the drawer to work properly.
+
+```jsx
+<!-- Template -->
+<svg class="template" width="250" height="200"></svg>
+```
+
+### 2. Add the hitbox :
+
+The hitbox is the area where you can click to select or drag the component, it is usually a `<rect />`but it can be any
+shape you want.
+
+**Important :** you need to fill it with the `fill` attribute and set a `component-hitbox`class.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <!-- Hitbox -->
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+</svg>
+```
+
+### 3. Add the icon container :
+
+The icon is the small icon that appears on the component, it is usually a `<svg></svg>` with a `component-icon` class.
+
+**Important :** you need to set the `id` attribute with the `{{id}}` syntax.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+
+    <!-- Icon container -->
+    <svg id="icon-{{id}}" class="component-icon" width="32" height="32"></svg>
+</svg>
+```
+
+*Note : the icon will be added and the viewBox will be set by the drawer, you don't need to add it manually.*
+
+### 4. Add the metadata you want display :
+In our case we want to display the name and the type of the component, so we add two `<text></text>` with the 
+`component-name` and `component-type` classes.
```

Comment:
In our case, we want to display the name and the type of the component. To do so, add two `<text></text>` with the `component-name` and `component-type` classes.

Reply:
Done